src-self-hosted/Module.zig

```diff
@@ -2682,64 +2681,96 @@ fn analyzeInstElemPtr(self: *Module, scope: *Scope, inst: *zir.Inst.ElemPtr) Inn
     return self.fail(scope, inst.base.src, "TODO implement more analyze elemptr", .{});
 }
 
-fn analyzeInstSub(self: *Module, scope: *Scope, inst: *zir.Inst.Sub) InnerError!*Inst {
-    return self.fail(scope, inst.base.src, "TODO implement analysis of sub", .{});
+fn analyzeInstBinOp(self: *Module, scope: *Scope, inst: *zir.Inst) InnerError!*Inst {
+    return switch (inst.tag) {
+        .add => self.analyzeInstArithmetic(scope, inst.cast(zir.Inst.Add).?),
+        .sub => self.analyzeInstArithmetic(scope, inst.cast(zir.Inst.Sub).?),
```

Comment:
These can be moved directly up to line 2173 - no need for control flow to take a detour through `analyzeInstBinOp`.